ducktape/cluster/node_container.py

```diff
@@ -112,16 +117,43 @@ def remove_spec(self, cluster_spec):
         :throws InsufficientResourcesError:     If there are not enough nodes in the NodeContainer.
                                                 Nothing will be removed unless enough are available.
         """
-        msg = self.attempt_remove_spec(cluster_spec)
-        if len(msg) > 0:
-            raise InsufficientResourcesError("Not enough nodes available to allocate. " + msg)
-        removed = []
+        err = ""
+        good = []
+        bad = []
         for os, node_specs in iteritems(cluster_spec.nodes.os_to_nodes):
             num_nodes = len(node_specs)
+            good_per_os = []
+            bad_per_os = []
             avail_nodes = self.os_to_nodes.get(os, [])
-            for i in range(0, num_nodes):
-                removed.append(avail_nodes.pop(0))
-        return removed
+            if len(avail_nodes) < num_nodes:
+                err = err + "%s nodes requested: %d. Total %s nodes available: %d" % \
+                  (os, num_nodes, os, len(avail_nodes))
+            else:
+                for i in range(0, num_nodes):
+                    node = avail_nodes.pop(0)
+                    if isinstance(node, RemoteAccount):
+                        try:
+                            if node.available():
+                                good_per_os.append(node)
+                            else:
+                                bad_per_os.append(node)
```

Comment:
is bad_per_os necessary, looks like you can append to bad directly.

Reply:
yeah, good point. Removed.